embark-consult.el

```diff
@@ -137,7 +137,9 @@ associated to an active minibuffer for a Consult command."
 (defun embark-consult--strip-suffix (string property)
   "Remove the final characters of STRING with the given PROPERTY."
   (let ((len (length string)))
-    (substring string 0 (or (previous-single-property-change len property string) len))))
+    (if (and (> len 0) (get-text-property (- len 1) property string))
+        (substring string 0 (or (previous-single-property-change len property string) len))
+      string)))
```

Comment:
Saves an unnecessary copy, I like it. This is the sort of optimization that in Vertico you'd have to weight against bumping up the line count by two... 😛

Reply:
I object to that! Vertico is highly optimized, this is what recently bumped the line count to 498 lines (https://github.com/minad/vertico/commit/a962a8b2b682bb99304aa022bb20d66a838fd1ea). @manuel-uberti observes the line count closely. I have to act wisely with the remaining two lines if I want to keep the users.

Reply:
Is deferring the highlighting really noticeable? When Clemens asked for the skip-highlighting variable in orderless I remember testing with Selectrum and not noticing any difference.